src/python/tensorflow_cloud/core/tests/integration/run_on_script_test.py

```diff
@@ -56,12 +56,228 @@ def delete_dir(self, path: Text) -> None:
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_auto_dist_strat_mwms_custom_img(self, mock_exit):
+        tfc.run(
+            entry_point=os.path.join(self.test_data_path, "mnist_example_using_fit.py"),
+            worker_count=1,
+            requirements_txt=os.path.join(self.test_data_path, "requirements.txt"),
+            docker_base_image="tensorflow/tensorflow:latest-gpu",
+        )
+        mock_exit.assert_called_once_with(0)
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_auto_dist_strat_mwms_bucket_build_custom_img(self, mock_exit):
+        tfc.run(
+            entry_point=os.path.join(self.test_data_path, "mnist_example_using_fit.py"),
+            worker_count=1,
+            requirements_txt=os.path.join(self.test_data_path, "requirements.txt"),
+            docker_image_bucket_name="TEST_BUCKET",
+            docker_base_image="tensorflow/tensorflow:latest-gpu",
+        )
+        mock_exit.assert_called_once_with(0)
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_none_dist_strat_mwms(self, mock_exit):
+        tfc.run(
+            entry_point=os.path.join(self.test_data_path, "mnist_example_using_ctl.py"),
+            worker_count=2,
+            requirements_txt=os.path.join(self.test_data_path, "requirements.txt"),
+        )
+        mock_exit.assert_called_once_with(0)
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_none_dist_strat_mwms_bucket_build(self, mock_exit):
+        tfc.run(
+            entry_point=os.path.join(self.test_data_path, "mnist_example_using_ctl.py"),
+            worker_count=2,
+            requirements_txt=os.path.join(self.test_data_path, "requirements.txt"),
+            docker_image_bucket_name="TEST_BUCKET",
+        )
+        mock_exit.assert_called_once_with(0)
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_none_dist_strat_mwms_custom_img(self, mock_exit):
+        tfc.run(
+            entry_point=os.path.join(self.test_data_path, "mnist_example_using_ctl.py"),
+            worker_count=2,
+            requirements_txt=os.path.join(self.test_data_path, "requirements.txt"),
+            docker_base_image="tensorflow/tensorflow:latest-gpu",
+        )
+        mock_exit.assert_called_once_with(0)
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_none_dist_strat_mwms_bucket_build_custom_img(self, mock_exit):
         tfc.run(
             entry_point=os.path.join(self.test_data_path, "mnist_example_using_ctl.py"),
-            distribution_strategy=None,
             worker_count=2,
             requirements_txt=os.path.join(self.test_data_path, "requirements.txt"),
+            docker_image_bucket_name="TEST_BUCKET",
+            docker_base_image="tensorflow/tensorflow:latest-gpu",
+        )
+        mock_exit.assert_called_once_with(0)
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_tpu_dist_strat_mwms(self, mock_exit):
```

Comment:
TPU is a distribution strategy just like multi worker strategy. Please rename to test_auto_tpu